app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,25 @@ def index
   def show
     @post = Post.find_by(id: params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @author = User.find_by(id: params[:user_id])
+    @post = @author.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_path(id: @post.author_id)
+    else
+      render :new, alert: ':( Cannot Create post retry again :('
+    end
+  end
+
```

Comment:
- [ ] The post should be created on behalf of the `current_user` defined in the application controller, not `user_id`. Kindly fix that.

- [ ] I am unable to create a post. You have validation errors related to comments and likes counter. Kindly define them before trying to save the post to resolve that.

  SUGGESTION:
  ```
  ...
  @post.likes_counter = 0
  @post.comments_counter = 0

  if @post.save
    redirect_to user_path(id: @post.author_id)
  else
  ...
  ```

  ![dido-1](https://user-images.githubusercontent.com/36542986/205067439-51522207-f975-4931-bcda-59c69d211965.jpg)